packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -825,8 +825,9 @@ String getAotBuildDirectory() {
 }
 
 /// Returns the asset build output directory.
-String getAssetBuildDirectory() {
-  return globals.fs.path.join(getBuildDirectory(), 'flutter_assets');
+String getAssetBuildDirectory([Config? config, FileSystem? fileSystem]) {
```

Comment:
when are these null? just tests? to me this pattern of optional parameters that fallback to grabbing globals is a code smell--but I can understand if always direct injecting would take a ridiculous amount of work.

Reply:
There are ~20 references to `getBuildDirectory` (this is excluding tests and references from the below functions). Each of the functions below that call this have a few references each. I could try to refactor this now, but it could take anywhere between 30 minutes and an entire day.

Reply:
This is good enough for me. can you add a TODO to make these required parameters then? I just want to avoid cargo-culting this pattern, if possible. But I think overall this is an improvement to the code--just want to make sure readers don't see this and think it's the blessed path :)

Reply:
Will do. I'm currently attempting the refactor to make these required, and I'll settle for the TODO if it starts taking any significant amount of time.

Reply:
sounds good to me. we'll have to pay this cost sooner or later...

Reply:
If we end up moving forward with your idea of a single context object that we DI everywhere, wouldn't that make this work here moot?

Reply:
Also this refactor would indeed be a significant amount of work _unless_ we are okay with adding more globals.dart imports.

Reply:
I suppose you're right, let's leave it for now. We'd have to touch all these functions, but it'd be a different refactor